### PR TITLE
Setup ESLint

### LINKED ISSUE #61: Set up code linter
Setting up a linter as `ESLint` will ensure the code consistency and detect potential bugs.

Maybe you can just enable http://codeclimate.com for now.

screen shot 2016-11-10 at 08 24 56

### PR DESCRIPTION
**Pull request by joaopapereira**
Fixes #61

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -3,11 +3,14 @@
...
   "main": "bin/server.js",
-  "engines" : { "node" : ">=6.0.0" },
+  "engines": {
+    "node": ">=6.0.0"
+  },
   "scripts": {
     "test": "cross-env NODE_ENV=test mocha -R spec test/*/*.spec.js",
     "start": "node bin/server.js",
     "debug": "node debug bin/server.js",
+    "lint": "./node_modules/.bin/eslint src/ --color",
```
Comment (arreche) [inline package.json]: `npm run` uses the `node_modules/.bin` path by default.

`"lint": "eslint src/ --color",`

File: src/story/story.controller.js [hunk 2]
```diff
@@ -13,9 +13,9 @@ exports.createStory = function (req, res, next) {
     var url = req.body.url;
     // console.log("Here is the request body: " + JSON.stringify(req.body))
     Story.create({"name": name, "size": size, "url": url}, function(err, story) {
-        if (err) throw err;
+        if (err) return next(err);
         // console.log('Story created!');
-        var id = story._id;
+        // var id = story._id;
```
Comment (arreche) [inline src/story/story.controller.js]: Commented out code can be deleted.

File: src/story/story.routes.js [hunk 3]
```diff
@@ -6,7 +6,7 @@ var voteController = require('../vote/vote.controller')
 
 //Middle ware that is specific to this router
 router.use(function timeLog(req, res, next) {
```
Comment (arreche) [inline src/story/story.routes.js]: This middleware is not doing anything so it could be deleted.

**Comment by arreche** [comment]
LGTM 👍

File: src/story/story.routes.js [hunk 4]
```diff
@@ -4,12 +4,6 @@ var router = express.Router();
 var controller = require('./story.controller');
 var voteController = require('../vote/vote.controller')
 
-//Middle ware that is specific to this router
```
Comment (arreche) [inline src/story/story.routes.js]: You are right. This is unrelated.

I understand we must be cautious introducing modifications to the current feature but in this case, this is dead/unused code.

**Comment by arreche** [comment]
👍

**Comment by tansaku** [comment]
hmm codeclimate not configured correctly ...

**Comment by arreche** [comment]
@tansaku It is configured now and it found issues in the code.
